src/features/Admin/Rooms/AddNewRoom/AddNewRoom.tsx

```diff
@@ -1,9 +1,407 @@
-import React from 'react'
+import { Button, Checkbox, Divider, FormControl, Grid, ListItemText, Paper, Select, TextField } from "@mui/material";
+import { Box, Container } from "@mui/system";
+import { useContext, useState, useEffect } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import Styles from "./AddNewRoom.module.scss";
+import { InputLabel, MenuItem } from "@mui/material";
+import axios from "axios";
+import { AuthContext } from "../../../../context/AuthContext";
+import { addroomsUrl } from "../../../../services/api";
+import { IAddRoom } from "../../../../interface/RoomInterface";
+import { faciRoomsUrl } from "./../../../../services/api";
+import DragDropFileUpload from "../../../../shared/DragDropFileUpload/DragDropFileUpload";
+const AddNewRoom: React.FC = () => {
+  const { requestHeaders } = useContext(AuthContext);
+
+  const navigate = useNavigate();
+
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [facilitiesList, setFacilitiesList] = useState([]);
+  const [selectedImage, setSelectedImage] = useState(null);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    formState: { errors },
+
+  } = useForm<IAddRoom>();
+
+  // Format Data
+  const appendToFormData = (data: IAddRoom) => {
+    const formData = new FormData();
+    formData.append("roomNumber", data["roomNumber"]);
+    formData.append("price", data["price"]);
+    formData.append("capacity", data["capacity"]);
+    formData.append("discount", data["discount"]);
+    // formData.append("facilities", data["facilities"]?.[0]);
+
+    if (Array.isArray(data.facilities)) {
+      data.facilities.forEach((facility) => {
+        formData.append("facilities[]", facility);
+      });
+    } else if (data.facilities) {
+      formData.append("facilities[]", data.facilities);
+    }
+
+    // formData.append("imgs", data["imgs"][0]);
+    return formData;
+  };
+// **** Cancel Button*******
+  const goBack = () => {
+    navigate(-1);
+  };
+  // *******Create New Room**********
+  const onSubmit: SubmitHandler<IAddRoom> = async (data: IAddRoom) => {
+    // setIsLoading(true)
+    const addFormData = appendToFormData(data);
+    axios
+      .post(`${addroomsUrl}`, addFormData, { headers: requestHeaders })
+      .then((response) => {
+        // setIsLoading(false)
+
```

Comment:
Don't forget to add toast messages, to indicate want is going wrong and what went well. This is very crucial for a better user experience.

Reply:
I'll add this in the next branch or create a branch specific to Toast context